ledger/participant-integration-api/src/test/suite/scala/platform/store/cache/EventsBufferSpec.scala

```diff
@@ -21,141 +21,231 @@ import scala.concurrent.duration.DurationInt
 import scala.concurrent.{Await, ExecutionContext, ExecutionContextExecutorService, Future}
 
 class EventsBufferSpec extends AnyWordSpec with Matchers with ScalaCheckDrivenPropertyChecks {
-  private val BufferElements = Vector(2, 3, 5, 8, 13).map(idx => idx -> idx * 2)
-  private val LastOffset = BufferElements.last._1
+  private val offsetIdx = Vector(2, 4, 6, 8, 10)
+  private val BeginOffset = offset(0L)
+  private val offsets @ Seq(offset1, offset2, offset3, offset4, offset5) =
+    offsetIdx.map(i => offset(i.toLong))
+  private val bufferElements @ Seq(_, entry2, entry3, entry4) =
+    offsets.zip(offsetIdx.map(_ * 2)).take(4)
+
+  private val LastOffset = offset4
+  private val IdentityFilter: Int => Option[Int] = Some(_)
 
   "push" when {
     "max buffer size reached" should {
       "drop oldest" in withBuffer(3L) { buffer =>
-        buffer.slice(0, LastOffset) shouldBe Prefix(BufferElements.drop(2))
-        buffer.push(21, 42)
-        buffer.slice(0, 21) shouldBe Prefix(BufferElements.drop(3) :+ 21 -> 42)
+        buffer.slice(BeginOffset, LastOffset, IdentityFilter) shouldBe Suffix(
```

Comment:
If we were to use `offset1` as the begin offset then bufferedStartExclusive may be `offset3` (due to double adjustment).